richard-kendra/server.js

```diff
@@ -0,0 +1,195 @@
+'use strict';
+
+const fs = require('fs');
+const express = require('express');
+const pg = require('pg');
+
+const PORT = process.env.PORT || 3000;
+const app = express();
+
+// Windows and Linux users: You should have retained the user/password from the pre-work for this course.
+// Your OS may require that your conString is composed of additional information including user and password.
+// const conString = 'postgres://kendragon:Zip9tape!@localhost:5432/kilovolt';
+const conString = 'postgres://rich:6641@localhost:5432/kilovolt';
+
+// Mac:
+// const conString = 'postgres://localhost:5432';
+
+const client = new pg.Client(conString);
+
+// REVIEW: Use the client object to connect to our DB.
+client.connect();
+
+
+// REVIEW: Install the middleware plugins so that our app can parse the request body
+app.use(express.json());
+app.use(express.urlencoded());
+app.use(express.static('./public'));
+
+
+// REVIEW: Routes for requesting HTML resources
+app.get('/new-article', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js, if any, is interacting with this particular piece of `server.js`? What part of CRUD, if any, is being enacted/managed by this particular piece of code?
+  // The following line of code matches #5 in the full-stack-diagram.png. It appears that no methods in article.js interact with this piece of server.js. This piece also is not a part of CRUD, because it's just serving the new.html page and does not do anything with the app's data in the database.
+  response.sendFile('new.html', {root: './public'});
+});
+
+
+// REVIEW: Routes for making API calls to use CRUD Operations on our database
+app.get('/articles', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // This line of code matches #3 & #5 in the diagram, which is making a query to the database and sending a response. This interacts with the method fetchAll() in article.js, which asks for all the article data. This corresponds to the "R" in CRUD, because it's retrieving data from the database.
```

Comment:
This is actually using 3 and 4. Querying the data and then sending it back to the server.